[Angular] Build fails if the name in the package.json does not match with the name of the project in angular.json

### Describe the bug

Hi @yatho @melloware,

since Quinoa 2.4.2 a build fails if the `package.json` name does not match the name of the Angular projects `angular.json`.
This is caused by PR #716 in lines
https://github.com/quarkiverse/quarkus-quinoa/blob/2.4.2/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java#L36
and
https://github.com/quarkiverse/quarkus-quinoa/blob/2.4.2/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java#L56-L59

This PR change was really necessary with the new Angular 17+ ESBuild + Vite builder and I appended `/browser` to my custom `quarkus.quinoa.build-dir` configuration option in the past manually with older Quinoa versions.

The build will also fail if the `package.json` name is not set at all, as this code then falls back to `quinoa`
```java
final String applicationName = packageJson.map(p -> p.getString("name")).orElse("quinoa");
```
which will most likely not exist as a project in the `angular.json`.

In the meantime, I aligned the names in my project, which fixes this build issue.

Exception:
```java
> Task :quarkusAppPartsBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusAppPartsBuild'.
> There was a failure while executing work items
   > A failure occurred while executing io.quarkus.gradle.tasks.worker.BuildWorker
      > io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        	[error]: Build step io.quarkiverse.quinoa.deployment.QuinoaProcessor#processBuild threw an exception: java.lang.NullPointerException: Cannot invoke "jakarta.json.JsonObject.getJsonObject(String)" because the return value of "jakarta.json.JsonObject.getJsonObject(String)" is null
        	at io.quarkiverse.quinoa.deployment.framework.override.AngularFramework$1.getBuilderUse(AngularFramework.java:58)
        	at io.quarkiverse.quinoa.deployment.framework.override.AngularFramework$1.buildDir(AngularFramework.java:38)
        	at io.quarkiverse.quinoa.deployment.QuinoaProcessor.processBuild(QuinoaProcessor.java:175)
        	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
        	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        	at java.base/java.lang.Thread.run(Thread.java:1583)
        	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
```

### Quinoa version

2.4.2

### Quarkus version

3.12.3

### Build / Runtime

Angular

### Package Manager

PNPM

### Steps to reproduce the behavior

Quarkus + Quinoa [Angular] + the `package.json` name does not match the `angular.json` project name.

### Expected behavior

If it really is the case that the `package.json` name and `angular.json` project name do not match:

1. Try to use the first key in the `angular.json` project Object as a fallback (or a better detection logic) where the `architect.build.builder` key exists
2. **Log a warning that the name does not match and that it should be fixed by the developer**

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java
@@ -29,23 +29,25 @@ public class AngularFramework extends GenericFramework {
         return new QuinoaConfigDelegate(super.override(originalConfig, packageJson, detectedDevScript, isCustomized)) {
             @Override
             public Optional<String> buildDir() {
-                final JsonObject angularJson = readAngularJson(originalConfig);
-                final JsonObject projectList = angularJson.getJsonObject("projects");
-                final JsonObject builder = projectList.values().stream()
-                        .map(JsonValue::asJsonObject)
-                        .filter(project -> "application".equals(project.getString("projectType")))
-                        .findFirst()
-                        .orElseThrow(() -> new RuntimeException(
-                                "Quinoa failed to determine which application must be started in the angular.json file."))
-                        .getJsonObject("architect")
-                        .getJsonObject("build");
+                return Optional.of(originalConfig.buildDir().orElseGet(() -> {
+                    final JsonObject angularJson = readAngularJson(originalConfig);
+                    final JsonObject projectList = angularJson.getJsonObject("projects");
+                    final JsonObject builder = projectList.values().stream()
+                            .map(JsonValue::asJsonObject)
+                            .filter(project -> "application".equals(project.getString("projectType")))
+                            .findFirst()
+                            .orElseThrow(() -> new RuntimeException(
+                                    "Quinoa failed to determine which application must be started in the angular.json file."))
+                            .getJsonObject("architect")
+                            .getJsonObject("build");
 
-                final String builderName = builder.getString("builder");
-                String fullBuildDir = builder.getJsonObject("options").getString("outputPath");
-                if (ANGULAR_DEVKIT_BUILD_ANGULAR_APPLICATION.equals(builderName)) {
-                    fullBuildDir = String.format("%s/browser", fullBuildDir);
-                }
-                return Optional.of(originalConfig.buildDir().orElse(fullBuildDir));
+                    final String builderName = builder.getString("builder");
+                    String fullBuildDir = builder.getJsonObject("options").getString("outputPath");
+                    if (ANGULAR_DEVKIT_BUILD_ANGULAR_APPLICATION.equals(builderName)) {
+                        fullBuildDir = String.format("%s/browser", fullBuildDir);
+                    }
+                    return fullBuildDir;
+                }));
             }
 
             private static JsonObject readAngularJson(QuinoaConfig configuration) {

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java
@@ -5,10 +5,7 @@ import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Optional;
 
-import jakarta.json.Json;
-import jakarta.json.JsonException;
-import jakarta.json.JsonObject;
-import jakarta.json.JsonReader;
+import jakarta.json.*;
 
 import io.quarkiverse.quinoa.deployment.config.PackageManagerCommandConfig;
 import io.quarkiverse.quinoa.deployment.config.QuinoaConfig;
@@ -32,12 +29,20 @@ public class AngularFramework extends GenericFramework {
         return new QuinoaConfigDelegate(super.override(originalConfig, packageJson, detectedDevScript, isCustomized)) {
             @Override
             public Optional<String> buildDir() {
-                // Angular builds a custom directory "dist/[appname]" or "dist/[appname]/browser" if it is build with application builder
-                final String applicationName = packageJson.map(p -> p.getString("name")).orElse("quinoa");
                 final JsonObject angularJson = readAngularJson(originalConfig);
-                final String builder = getBuilderUse(angularJson, applicationName);
-                String fullBuildDir = String.format("%s/%s", getDefaultBuildDir(), applicationName);
-                if (ANGULAR_DEVKIT_BUILD_ANGULAR_APPLICATION.equals(builder)) {
+                final JsonObject projectList = angularJson.getJsonObject("projects");
+                final JsonObject builder = projectList.values().stream()
+                        .map(JsonValue::asJsonObject)
+                        .filter(project -> "application".equals(project.getString("projectType")))
+                        .findFirst()
+                        .orElseThrow(() -> new RuntimeException(
+                                "Quinoa failed to determine which application must be started in the angular.json file."))
+                        .getJsonObject("architect")
+                        .getJsonObject("build");
+
+                final String builderName = builder.getString("builder");
+                String fullBuildDir = builder.getJsonObject("options").getString("outputPath");
+                if (ANGULAR_DEVKIT_BUILD_ANGULAR_APPLICATION.equals(builderName)) {
                     fullBuildDir = String.format("%s/browser", fullBuildDir);
                 }
                 return Optional.of(originalConfig.buildDir().orElse(fullBuildDir));
@@ -51,11 +56,6 @@ public class AngularFramework extends GenericFramework {
                 } catch (IOException | JsonException e) {
                     throw new RuntimeException("Quinoa failed to read the angular.json file. %s", e);
                 }
-            }
-
-            private static String getBuilderUse(JsonObject angularJson, String applicationName) {
-                return angularJson.getJsonObject("projects").getJsonObject(applicationName)
-                        .getJsonObject("architect").getJsonObject("build").getString("builder");
             }
 
             @Override

--- a/integration-tests/src/main/ui-angular-esbuild/package.json
+++ b/integration-tests/src/main/ui-angular-esbuild/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "quinoa-app",
+  "name": "quinoa",
   "version": "0.0.0",
   "scripts": {
     "ng": "ng",

